### DISCUSSION TITLE
Duplicates of Same Version

### DISCUSSION THREAD

**Discussion by shallow-beach** [coreybutler/nvm-windows#725] [Support] [answered]
wondering if there is a way to name (and duplicate) our version of node.js, such that I may switch between 'profiles'?

I am in a university course that restricts which utilities we are allowed, but I also want to explore what's available on my own. I would like a 'class' profile and a 'default' profile - preferably I could selectively update and add global utilities to the default profile at will, while leaving the class profile clean and unupdated. 

it seems the original nvm author left this function out for minimality (from github thread), because all one needs to do is duplicate the folder with a letter appended. with nvm-windows, anything other than very specifically the ab.c.d patch folder name, ie not ab.c.de or ab.c.d.e, returns an error "unrecognized patch 'c.de'/'c.d.e'".  for clarity, with the current edition, trying to duplicate the folder 'v17.3.1' to a 'v17.3.1b' gives an error when entering 'nvm use v17.3.1b' that says "unrecognized patch '3.1b'. the same applies when trying 'v17.3.1.1'. with an "unrecognized '3.1.1'".

thanks!

**Comment by coreybutler** [comment, marked as answer]
This is not currently possible with NVM4W. However; the concept of profiles is something I'm working on for a related product.

That said, it is not as simple as having another copy of a folder. Profiles need to be maintained with each version - tracked independently. For just a profile or two, an addition to the settings file may suffice. However; it gets exponentially more difficult to track versions with more profiles. This can lead to significant (i.e. unmanageable) bloat. This causes problems in environments where modules are stored on NAS/SAN drives or where user profiles are replicated.

Furthermore, this is not an npm or yarn management tool - it's a node version manager. I hear the use case loud and clear, but this is not the right tool to implement it. Expect to see these kinds of features in a future project.
